### PR TITLE
add CUIWnd::Flash

### PR DESCRIPTION
**Pull request by TheSilentOne1**
If Flash() is called, the window will periodically brighten up. The duration of the period is specified by SetFlashDuration(). The purpose of this is to have a method to draw the player's attention to a window, e. g. to the message window if a message is received.
In preparation of diplomacy UI changes.

### CODE REVIEW

File: UI/CUIWnd.cpp [hunk 1]
```diff
@@ -318,7 +318,10 @@ void CUIWnd::Render() {
         glDrawArrays(GL_LINE_LOOP,      m_buffer_indices[0].first, m_buffer_indices[0].second);
 
     } else {
-        glColor(ClientUI::WndColor());
+        if (m_flashing && GG::GUI::GetGUI()->Ticks() % (m_flash_duration * 2) > m_flash_duration)
+            glColor(GG::LightColor(ClientUI::WndColor()));
```
Comment (geoffthemedio) [inline UI/CUIWnd.cpp]: Won't work well if the WndColor is already fairly light...

File: UI/CUIWnd.cpp [hunk 2]
```diff
@@ -318,7 +318,10 @@ void CUIWnd::Render() {
...
+        else
+            glColor(ClientUI::WndColor());
         glDrawArrays(GL_TRIANGLE_FAN,   m_buffer_indices[1].first, m_buffer_indices[1].second);
         glColor(ClientUI::WndOuterBorderColor());
```
Comment (geoffthemedio) [inline UI/CUIWnd.cpp]: Should probably also flash the border oulines...

Comment (geoffthemedio) [inline UI/CUIWnd.cpp]: Should probably also flash the border outlines

File: UI/CUIWnd.cpp (hunk 1 above)
Comment (TheSilentOne1) [reply UI/CUIWnd.cpp]: Did some test with bright window background and I don't really think it is an issue. Did you have something specific in mind?

File: UI/CUIWnd.cpp (hunk 2 above)
Comment (TheSilentOne1) [reply UI/CUIWnd.cpp]: Sounds good.

File: UI/CUIWnd.cpp (hunk 1 above)
Comment (geoffthemedio) [reply UI/CUIWnd.cpp]: Idea would be to make the colour darker instead of lighter if it's already more than ~70% white.

**Comment by geoffthemedio** [comment]
Best I can tell, the CI errors can be ignored...

File: UI/CUIWnd.cpp (hunk 1 above)
Comment (TheSilentOne1) [reply UI/CUIWnd.cpp]: Sounds okay to me, currently I'm hindered, I can add it later.

**Pull request merged**